qa/evil-tests/src/test/java/org/elasticsearch/common/logging/EvilLoggerTests.java

```diff
@@ -141,6 +141,18 @@ public void testPrefixLogger() throws IOException, IllegalAccessException, UserE
         }
     }
 
+    public void testPrefixLoggerMarkersCanBeCollected() throws IOException, UserException {
+        setupLogging("prefix");
+
+        for (int i = 0; i < 1 << 20; i++) {
```

Comment:
Maybe move the limit out to a local, with a comment on the reason for the size (I presume to ensure System.gc() actually collects something below).